client/client.go

```diff
@@ -281,21 +280,142 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
-func dialOpenSSL(conf Config, network string, addr string) (net.Conn, error) {
-	contextSSL, err := openssl.NewCtx()
-	// probably should consider reusing the context, but then we
-	// have to propagate it with every request.
-	err = contextSSL.LoadVerifyLocations(conf.ServerCert, "")
+// nrOfSystemCertsFound simply returns the number of certificates found in
+// 'certDir'. The only reason this is needed, is so that the user can be
+// notified if the system certificate directory is empty, since this is not done
+// by the OpenSSL wrapper, in the same manner it was done by the Go standard
+// library. OpenSSL loads the trust chain on a dial. This system cert, and
+// server cert path are set by the 'SetDefaultVerifyPaths' and
+// 'LoadVerifyLocations' respectively.
+func nrOfSystemCertsFound(certDir string) (int, error) {
+	sysCertsFound := 0
+	files, err := ioutil.ReadDir(certDir)
 	if err != nil {
-		return nil, err
+		return 0, fmt.Errorf("Failed to read the OpenSSL default directory (%s). Err %v", certDir, err.Error())
 	}
+	for _, certFile := range files {
+		certBytes, err := ioutil.ReadFile(path.Join(certDir, certFile.Name()))
+		if err != nil {
+			log.Debugf("Failed to read the certificate file for the HttpsClient. Err %v", err.Error())
+			continue
+		}
+
+		certs := openssl.SplitPEM(certBytes)
+		if len(certs) == 0 {
+			log.Debugf("No PEM certificate found in '%s'", certFile)
+			continue
+		}
+		first, certs := certs[0], certs[1:]
+		_, err = openssl.LoadCertificateFromPEM(first)
+		if err != nil {
+			log.Debug(err.Error())
+		} else {
+			sysCertsFound += 1
+		}
+	}
+	return sysCertsFound, nil
+}
+
+func loadServerTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
+	defaultCertDir, err := openssl.GetDefaultCertificateDirectory()
+	if err != nil {
+		return ctx, errors.New("Failed to get the default OpenSSL certificate directory. Please verify the OpenSSL setup")
+	}
+	sysCertsFound, err := nrOfSystemCertsFound(defaultCertDir)
+	if err != nil {
+		log.Warnf("Failed to list the system certificates with error: %v", err)
+	}
+
+	// Set the default system certificate path for this OpenSSL context
+	err = ctx.SetDefaultVerifyPaths()
+	if err != nil {
+		return ctx, fmt.Errorf("Failed to set the default OpenSSL directory. OpenSSL error code: %s", err.Error())
+	}
+	// Load the server certificate into the OpenSSL context
+	err = ctx.LoadVerifyLocations(conf.ServerCert, "")
+	if err != nil {
+		if strings.Contains(err.Error(), "No such file or directory") {
+
+			log.Errorf(errMissingServerCertF, conf.ServerCert)
```

Comment:
This was logged at `Warnf` level in the old client, and I think we should keep that, since this is the normal operating procedure for example for Hosted Mender. If we really don't have *any* certificate, we will hit the other error below.